From 91b52118a3f7383f2ad65ebab87498943f241ee1 Mon Sep 17 00:00:00 2001
From: Yiming Cao <cyiming@codeaurora.org>
Date: Tue, 10 May 2016 15:22:19 +0800
Subject: [PATCH] mm-core: migration for agl

1, stripe android dependence
2, disable vpp property by default
3, remove dsmode and secure omx components by default
4, upgrade log control and log output is disabled by default

Change-Id: Ic99dc40dbb413800ad85dbfbfc3a73082a38caeb
Signed-off-by: Yiming Cao <cyiming@codeaurora.org>
---
 Makefile.am                                  |  5 ++
 configure.ac                                 | 71 ++++++++++++++++++++++++++++
 mm-core/Makefile.am                          | 59 +++++++++++++++++++++++
 mm-core/inc/qc_omx_msg.h                     | 29 +++++++++---
 mm-core/src/common/qc_omx_core.c             | 16 ++++++-
 mm-core/src/msm8996/registry_table_android.c | 24 +++++++++-
 6 files changed, 195 insertions(+), 9 deletions(-)
 create mode 100644 Makefile.am
 create mode 100644 configure.ac
 create mode 100755 mm-core/Makefile.am

diff --git a/Makefile.am b/Makefile.am
new file mode 100644
index 0000000..2f8fe09
--- /dev/null
+++ b/Makefile.am
@@ -0,0 +1,5 @@
+# Makefile.am - Automake script for mm-omxvideo
+#
+ACLOCAL_AMFLAGS = -I m4
+
+SUBDIRS = mm-core
diff --git a/configure.ac b/configure.ac
new file mode 100644
index 0000000..6ff899c
--- /dev/null
+++ b/configure.ac
@@ -0,0 +1,71 @@
+#                                               -*- Autoconf -*-
+
+# configure.ac -- Autoconf script for mm-omxvideo
+#
+
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ(2.61)
+AC_INIT([omxvideo],
+	1.1.0)
+AM_INIT_AUTOMAKE([-Wall -Werror gnu foreign subdir-objects])
+AM_MAINTAINER_MODE
+AC_CONFIG_HEADER([config.h])
+AC_CONFIG_MACRO_DIR([m4])
+
+#release versioning
+OMXVIDEO_MAJOR_VERSION=1
+OMXVIDEO_MINOR_VERSION=1
+OMXVIDEO_MICRO_VERSION=0
+
+OMXVIDEO_LIBRARY_VERSION=$OMXVIDEO_MAJOR_VERSION:$OMXVIDEO_MINOR_VERSION:$OMXVIDEO_MICRO_VERSION
+AC_SUBST(OMXVIDEO_LIBRARY_VERSION)
+
+# Checks for programs.
+AM_PROG_AR
+AC_PROG_CC
+AC_PROG_CPP
+AC_PROG_CXX
+AM_PROG_CC_C_O
+AC_PROG_LIBTOOL
+AC_PROG_AWK
+AC_PROG_INSTALL
+AC_PROG_LN_S
+AC_PROG_MAKE_SET
+
+AC_ARG_ENABLE([target-msm8960],
+	AC_HELP_STRING([--enable-target-msm8960],
+		[Enable conditional compile for target msm8960 [default=no]]),
+	[target_msm8960="${enableval}"],
+	target_msm8960=no)
+
+AC_ARG_ENABLE([target-msm8974],
+	AC_HELP_STRING([--enable-target-msm8974],
+		[Enable conditional compile for target msm8974 [default=no]]),
+	[target_msm8974="${enableval}"],
+	target_msm8974=no)
+
+AM_CONDITIONAL(TARGET_MSM8960, test "x$target_msm8960" = "xyes")
+AM_CONDITIONAL(TARGET_MSM8974, test "x$target_msm8974" = "xyes")
+
+AC_ARG_WITH([sanitized-headers],
+	[AS_HELP_STRING([--with-sanitized-headers=DIR],[location of the sanitized Linux headers])],
+	[CPPFLAGS="$CPPFLAGS -I$withval -I${withval}/C2D"])
+
+AC_ARG_WITH([common-includes],
+	[AS_HELP_STRING([--with-common-include=DIR],[location of common headers])],
+	[CPPFLAGS="$CPPFLAGS -I$withval -I${withval}/C2D"])
+
+AC_ARG_WITH([libhardware-headers],
+	[AS_HELP_STRING([--with-libhardware-headers=DIR],[location of the libhardware headers])],
+	[CPPFLAGS="$CPPFLAGS -I${withval}/modules/gralloc"])
+
+AC_SUBST([CFLAGS])
+AC_SUBST([CPPFLAGS])
+AC_SUBST([CC])
+
+AC_CONFIG_FILES([     \
+	Makefile \
+	mm-core/Makefile
+	])
+AC_OUTPUT
diff --git a/mm-core/Makefile.am b/mm-core/Makefile.am
new file mode 100755
index 0000000..3439be7
--- /dev/null
+++ b/mm-core/Makefile.am
@@ -0,0 +1,59 @@
+# sources and intermediate files are separated
+
+AM_CFLAGS = -Wall
+AM_CFLAGS += -Wundef
+AM_CFLAGS += -Wstrict-prototypes
+AM_CFLAGS += -Wno-trigraphs
+AM_CFLAGS += -g -O3
+AM_CFLAGS += -O0 -fno-inline -fno-short-enums
+AM_CFLAGS += -fPIC
+
+
+AM_CPPFLAGS = -U_ENABLE_QC_MSG_LOG_
+AM_CPPFLAGS += -DVERBOSE
+#AM_CPPFLAGS += -D_DSMODE_CMP_ -D_SECURE_CMP_
+#AM_CPPFLAGS += -DVPP_PROPERTY
+AM_CPPFLAGS += -I$(top_srcdir)/mm-core/inc
+AM_CPPFLAGS += -I$(top_srcdir)/mm-core/src/common
+
+h_sources =inc/OMX_Audio.h
+h_sources +=inc/OMX_Component.h
+h_sources +=inc/OMX_ContentPipe.h
+h_sources +=inc/OMX_Core.h
+h_sources +=inc/OMX_Image.h
+h_sources +=inc/OMX_Index.h
+h_sources +=inc/OMX_IVCommon.h
+h_sources +=inc/OMX_Other.h
+h_sources +=inc/OMX_QCOMExtns.h
+h_sources +=inc/OMX_Types.h
+h_sources +=inc/OMX_Video.h
+h_sources +=inc/qc_omx_common.h
+h_sources +=inc/qc_omx_component.h
+h_sources +=inc/qc_omx_msg.h
+h_sources +=inc/QOMX_AudioExtensions.h
+h_sources +=inc/QOMX_AudioIndexExtensions.h
+h_sources +=inc/OMX_CoreExt.h
+h_sources +=inc/QOMX_CoreExtensions.h
+h_sources +=inc/QOMX_FileFormatExtensions.h
+h_sources +=inc/QOMX_IVCommonExtensions.h
+h_sources +=inc/QOMX_SourceExtensions.h
+h_sources +=inc/QOMX_VideoExtensions.h
+h_sources +=inc/OMX_IndexExt.h
+h_sources +=inc/QOMX_StreamingExtensions.h
+
+c_sources =src/common/omx_core_cmp.cpp
+c_sources +=src/common/qc_omx_core.c
+
+#if TARGET_MSM8960
+#c_sources +=src/8960/qc_registry_table.c
+#endif
+
+if TARGET_MSM8974
+#c_sources +=src/msm8996/registry_table.c
+c_sources +=src/msm8996/registry_table_android.c
+endif
+
+include_HEADERS = $(h_sources)
+lib_LTLIBRARIES = libOmxCore.la
+libOmxCore_la_SOURCES = $(c_sources)
+libOmxCore_la_LDFLAGS = -ldl -lrt -lpthread -shared
diff --git a/mm-core/inc/qc_omx_msg.h b/mm-core/inc/qc_omx_msg.h
index deb0ab7..63a285b 100644
--- a/mm-core/inc/qc_omx_msg.h
+++ b/mm-core/inc/qc_omx_msg.h
@@ -1,5 +1,5 @@
 /*--------------------------------------------------------------------------
-Copyright (c) 2009, The Linux Foundation. All rights reserved.
+Copyright (c) 2009, 2016, The Linux Foundation. All rights reserved.
 
 Redistribution and use in source and binary forms, with or without
 modification, are permitted provided that the following conditions are met:
@@ -73,14 +73,29 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
         #define DEBUG_PRINT       LOGI
         #define DEBUG_DETAIL      LOGV
     #else
-        #define DEBUG_PRINT_ERROR printf
-        #define DEBUG_PRINT       printf
-        #define DEBUG_DETAIL      printf
+#include <stdlib.h>
+        enum {
+          PRIO_ERROR=1,
+          PRIO_INFO=1,
+          PRIO_HIGH=2,
+          PRIO_LOW=4
+        };
+        #define DEBUG_PRINT_CTL(level, fmt, args...)  \
+          do {                                        \
+            char *ptr = getenv("OMX_DEBUG_LEVEL");    \
+            if ( level & (ptr?atoi(ptr):0) )          \
+               printf(""fmt"\n", ##args);} while(0)
+        #define DEBUG_PRINT_ERROR(fmt,args...) \
+             DEBUG_PRINT_CTL(PRIO_ERROR, fmt, ##args)
+        #define DEBUG_PRINT(fmt,args...)       \
+             DEBUG_PRINT_CTL(PRIO_LOW, fmt, ##args)
+        #define DEBUG_DETAIL(fmt,args...)      \
+             DEBUG_PRINT_CTL(PRIO_HIGH, fmt, ##args)
     #endif // _ANDROID_
 #else
-    #define DEBUG_PRINT_ERROR
-    #define DEBUG_PRINT
-    #define DEBUG_DETAIL
+    #define DEBUG_PRINT_ERROR(...)
+    #define DEBUG_PRINT(...)
+    #define DEBUG_DETAIL(...)
 #endif // _ENABLE_QC_MSG_LOG_
 
 #endif // _QC_OMX_MSG_H_
diff --git a/mm-core/src/common/qc_omx_core.c b/mm-core/src/common/qc_omx_core.c
index 493cb58..32355d0 100644
--- a/mm-core/src/common/qc_omx_core.c
+++ b/mm-core/src/common/qc_omx_core.c
@@ -1,5 +1,5 @@
 /*--------------------------------------------------------------------------
-Copyright (c) 2009, 2015, The Linux Foundation. All rights reserved.
+Copyright (c) 2009, 2015-2016, The Linux Foundation. All rights reserved.
 
 Redistribution and use in source and binary forms, with or without
 modification, are permitted provided that the following conditions are met:
@@ -47,7 +47,14 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include "qc_omx_core.h"
 #include "omx_core_cmp.h"
+#ifdef _ANDROID_
 #include <cutils/properties.h>
+#else
+#define PROPERTY_VALUE_MAX 92
+#include <glib.h>
+#define strlcpy g_strlcpy
+#define strlcat g_strlcat
+#endif
 
 extern omx_core_cb_type core[];
 extern const unsigned int SIZE_OF_CORE;
@@ -457,12 +464,18 @@ OMX_GetHandle(OMX_OUT OMX_HANDLETYPE*     handle,
     }
     if(cmp_index >= 0)
     {
+#ifdef VPP_PROPERTY
       char value[PROPERTY_VALUE_MAX];
       DEBUG_PRINT("getting fn pointer\n");
 
       // Load VPP omx component for decoder if vpp
       // property is enabled
+#ifdef _ANDROID_
       if ((property_get("media.vpp.enable", value, NULL))
+#else
+      value[0] = '\0';  //vpp property is disabled
+      if ( 1
+#endif
            && (!strcmp("1", value) || !strcmp("true", value))) {
         DEBUG_PRINT("VPP property is enabled");
         if (!strcmp(core[cmp_index].so_lib_name, "libOmxVdec.so")) {
@@ -475,6 +488,7 @@ OMX_GetHandle(OMX_OUT OMX_HANDLETYPE*     handle,
           }
         }
       }
+#endif
 
        // dynamically load the so
       core[cmp_index].fn_ptr =
diff --git a/mm-core/src/msm8996/registry_table_android.c b/mm-core/src/msm8996/registry_table_android.c
index 1e010f4..d481e02 100644
--- a/mm-core/src/msm8996/registry_table_android.c
+++ b/mm-core/src/msm8996/registry_table_android.c
@@ -1,5 +1,5 @@
 /*--------------------------------------------------------------------------
-Copyright (c) 2014-2015, The Linux Foundation. All rights reserved.
+Copyright (c) 2014-2016, The Linux Foundation. All rights reserved.
 
 Redistribution and use in source and binary forms, with or without
 modification, are permitted provided that the following conditions are
@@ -55,6 +55,7 @@ omx_core_cb_type core[] =
       "video_decoder.avc"
     }
   },
+#ifdef _DSMODE_CMP_
   {
     "OMX.qcom.video.decoder.avc.dsmode",
     NULL, // Create instance function
@@ -71,6 +72,8 @@ omx_core_cb_type core[] =
       "video_decoder.avc"
     }
   },
+#endif
+#ifdef _SECURE_CMP_
   {
     "OMX.qcom.video.decoder.avc.secure",
     NULL, // Create instance function
@@ -87,6 +90,8 @@ omx_core_cb_type core[] =
       "video_decoder.avc"
     }
   },
+#endif
+#if defined (_DSMODE_CMP_) && defined (_SECURE_CMP_)
   {
     "OMX.qcom.video.decoder.avc.secure.dsmode",
     NULL, // Create instance function
@@ -103,6 +108,7 @@ omx_core_cb_type core[] =
       "video_decoder.avc"
     }
   },
+#endif
 
   {
     "OMX.qcom.video.decoder.divx4",
@@ -168,6 +174,7 @@ omx_core_cb_type core[] =
       "video_decoder.mpeg4"
     }
   },
+#ifdef _SECURE_CMP_
   {
     "OMX.qcom.video.decoder.mpeg4.secure",
     NULL, // Create instance function
@@ -184,6 +191,7 @@ omx_core_cb_type core[] =
       "video_decoder.mpeg4"
     }
   },
+#endif
   {
     "OMX.qcom.video.decoder.mpeg2",
     NULL, // Create instance function
@@ -200,6 +208,7 @@ omx_core_cb_type core[] =
       "video_decoder.mpeg2"
     }
   },
+#ifdef _SECURE_CMP_
   {
     "OMX.qcom.video.decoder.mpeg2.secure",
     NULL, // Create instance function
@@ -216,6 +225,7 @@ omx_core_cb_type core[] =
       "video_decoder.mpeg2"
     }
   },
+#endif
   {
     "OMX.qcom.video.decoder.vc1",
     NULL, // Create instance function
@@ -232,6 +242,7 @@ omx_core_cb_type core[] =
       "video_decoder.vc1"
     }
   },
+#ifdef _SECURE_CMP_
   {
     "OMX.qcom.video.decoder.vc1.secure",
     NULL, // Create instance function
@@ -248,6 +259,7 @@ omx_core_cb_type core[] =
       "video_decoder.vc1"
     }
   },
+#endif
   {
     "OMX.qcom.video.decoder.wmv",
     NULL, // Create instance function
@@ -264,6 +276,7 @@ omx_core_cb_type core[] =
       "video_decoder.vc1"
     }
   },
+#ifdef _SECURE_CMP_
   {
     "OMX.qcom.video.decoder.wmv.secure",
     NULL, // Create instance function
@@ -280,6 +293,7 @@ omx_core_cb_type core[] =
       "video_decoder.vc1"
     }
   },
+#endif
   {
     "OMX.qcom.video.decoder.h263",
     NULL, // Create instance function
@@ -312,6 +326,7 @@ omx_core_cb_type core[] =
       "video_decoder.hevc"
     }
   },
+#ifdef _SECURE_CMP_
   {
     "OMX.qcom.video.decoder.hevc.secure",
     NULL,   // Create instance function
@@ -328,6 +343,7 @@ omx_core_cb_type core[] =
       "video_decoder.hevc"
     }
   },
+#endif
   {
     "OMX.qcom.video.decoder.vp8",
     NULL, // Create instance function
@@ -360,6 +376,7 @@ omx_core_cb_type core[] =
       "video_decoder.vp9"
     }
   },
+#ifdef _SECURE_CMP_
   {
     "OMX.qcom.video.decoder.vp9.secure",
     NULL, // Create instance function
@@ -376,6 +393,7 @@ omx_core_cb_type core[] =
       "video_decoder.vp9"
     }
   },
+#endif
   {
     "OMX.qcom.video.encoder.mpeg4",
     NULL, // Create instance function
@@ -424,6 +442,7 @@ omx_core_cb_type core[] =
       "video_encoder.avc"
     }
   },
+#ifdef _SECURE_CMP_
   {
     "OMX.qcom.video.encoder.avc.secure",
     NULL, // Create instance function
@@ -440,6 +459,7 @@ omx_core_cb_type core[] =
       "video_encoder.avc"
     }
   },
+#endif
   {
     "OMX.qcom.video.encoder.vp8",
     NULL, // Create instance function
@@ -472,6 +492,7 @@ omx_core_cb_type core[] =
       "video_encoder.hevc"
     }
   },
+#ifdef _SECURE_CMP_
   {
     "OMX.qcom.video.encoder.hevc.secure",
     NULL,   // Create instance function
@@ -488,6 +509,7 @@ omx_core_cb_type core[] =
       "video_encoder.hevc"
     }
   },
+#endif
   {
     "OMX.qcom.audio.decoder.Qcelp13",
     NULL, // Create instance function
-- 
1.9.1

